proposals/PluginsAudit2015/audit.md

```diff
@@ -0,0 +1,223 @@
+* Open Pull Requests: 20
+* Recommendation: Keep. Continue merging PRs and avoiding adding too many new features
+
+## Console
+* 1 Month Downloads: 45894
+* Open Issues: 5
+* Open Pull Requests: 1
+* console output should show up in Xcode by default. 
+* Recommendation: Move functionality back into platforms
+
+## Contacts
+* 1 Month Downloads: 6096
+* Open Issues: 71
+* Open Pull Requests: 10
+* Spec: http://www.w3.org/TR/contacts-api/ (discontinued)
+* API needs to be redone. Currently brittle and many platform specific quirks
+* Recommendation: Keep. Leave it as is and revist if new Contacts API is published by W3C. Api does need to be redone, but low priority.
+
+## Device
+* 1 Month Downloads: 53695
+* Open Issues: 7
+* Open Pull Requests: 1
+* Refactor: Update interface to be async?
+* Maybe keep it the same
+* Recommendation: Keep. Discuss if it makes sense to add it to hello world template as a default plugin. Add it to phonegap template. Add it as a dependency on tests that need it (Device-motion). Discuss if switching api async makes sense.
+
+## Device Orientation (compass)
+* 1 Month Downloads: 3823
+* Open Issues: 8
+* Open Pull Requests: 3
+* No Spec
+* maybe combine device-motion + compass?
+* Needs better API
+* rename to cordova plugin compass? 
+* Recommendation: Keep. Consider renaming plugin to cordova-plugin-compass. Needs improved API but low priority.
+
+## Device Motion (Accelerometer)
+* 1 Month Downloads: 3886
+* Open Issues: 8
+* Open Pull Requests: 0
+* spec http://www.w3.org/TR/orientation-event/
+* Starting to become available in browsers http://caniuse.com/#search=orientation 
+* rename to cordova-plugin-device-orientation-motion?
+* low priority: polish code, update code to remove accelerometer references. 
+* Recommendation: Keep. Code needs a cleanup but low priority.
+
+## Dialogs
+* 1 Month Downloads: 12456
+* Open Issues: 13
+* Open Pull Requests: 17
+* Using custom styled divs should be used instead of alerts
+* In HTML 5.1 there is a proposed dialog element which actually seems pretty good. http://www.w3.org/html/wg/drafts/html/master/single-page.html#the-dialog-element 
+* looks good on android
+* looks ugly on ios
+* currently not sync which is how default alerts work 
+* window.alert still available but says index.html
+* Recommendation: Deprecate! Encourage use of HTML/CSS/JS divs instead.
+
+## File
+* 1 Month Downloads: 21252
+* Open Issues: 60
+* Open Pull Requests: 15
+* phonegap-plugin-file or phonegap-plugin-fs. Consumes cordova-plugin-file. Create downstream version with a better interface. Current one is based on standard which is ugly. (rip shelljs file ops)
+* Can we treat some urls consistency across all platforms? Possibly use Flex Air package for inspiration.
+* ex: app:, temp:, 
+* cdvfile wasn’t consistent enough across platforms
+* Recommendation: Keep. Work on creating more consistency across platforms. Especially with cdvfile. Create phonegap-plugin-file with improved api.
```

Comment:
Could use File wrapping in app-harness as one point of inspiration: https://github.com/apache/cordova-app-harness/blob/master/www/cdvah/js/ResourcesLoader.js#L149